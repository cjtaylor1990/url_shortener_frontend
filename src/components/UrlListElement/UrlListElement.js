import React from 'react';
import './UrlListElement.css';

class UrlListElement extends React.Component {

	render() {
    	return(
			<div data-testid="urlListElement" className="urlListElement">
				<div data-testid="urlListElementFull" className="fullUrl">{this.props.urlPair.fullUrl}</div>
				<div data-testid="urlListElementShort" className="shortUrl">{this.props.urlPair.shortUrl}</div>
			</div>
		);
	};

};

export default UrlListElement;
