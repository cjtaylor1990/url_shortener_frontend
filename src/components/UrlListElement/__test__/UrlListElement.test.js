import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import UrlListElement from '../UrlListElement.js';

afterEach(cleanup);

const fullUrl = 'http//www.userslongurl.com';
const shortKey = 'shorturl'
const shortUrl = 'http//www.urlshortener.com/' + shortKey;
const urlPair = {
    shortUrl: shortUrl,
    fullUrl: fullUrl
};

describe('After being passed a URL pair, the UrlListElement initially', () => {
    it('renders without crashing', () => {

        const div = document.createElement('div');
            
        ReactDOM.render(<UrlListElement urlPair={urlPair} />, div);
            
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<UrlListElement urlPair={urlPair} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the full and short URL pair', () => {
        const { getByTestId } = render(<UrlListElement urlPair={urlPair} />);

        const renderedFullUrl = getByTestId('urlListElementFull');
        const renderedShortUrl = getByTestId('urlListElementShort');

        expect(renderedFullUrl).toHaveTextContent(fullUrl);
        expect(renderedShortUrl).toHaveTextContent(shortUrl);
    });
    
});

