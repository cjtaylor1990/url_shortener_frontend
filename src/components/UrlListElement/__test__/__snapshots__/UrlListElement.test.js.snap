// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`After being passed a URL pair, the UrlListElement initially matches snapshot 1`] = `
<div
  className="urlListElement"
  data-testid="urlListElement"
>
  <div
    className="fullUrl"
    data-testid="urlListElementFull"
  >
    http//www.userslongurl.com
  </div>
  <div
    className="shortUrl"
    data-testid="urlListElementShort"
  >
    http//www.urlshortener.com/shorturl
  </div>
</div>
`;
